airflow/migrations/versions/0152_2_10_3_fix_dag_schedule_dataset_alias_reference_naming.py

```diff
@@ -0,0 +1,119 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Rename dag_schedule_dataset_alias_reference constraint names.
+
+Revision ID: 5f2621c13b39
+Revises: 22ed7efa9da2
+Create Date: 2024-10-25 04:03:33.002701
+
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from alembic import op
+from sqlalchemy import inspect
+
+# revision identifiers, used by Alembic.
+revision = "5f2621c13b39"
+down_revision = "22ed7efa9da2"
+branch_labels = None
+depends_on = None
+airflow_version = "2.10.3"
+
+if TYPE_CHECKING:
+    from alembic.operations.base import BatchOperations
+    from sqlalchemy.sql.elements import conv
+
+
+def _rename_fk_constraint(
+    *,
+    batch_op: BatchOperations,
+    original_name: str | conv,
+    new_name: str | conv,
+    referent_table: str,
+    local_cols: list[str],
+    remote_cols: list[str],
+    ondelete: str,
+) -> None:
+    batch_op.drop_constraint(original_name, type_="foreignkey")
+    batch_op.create_foreign_key(
+        constraint_name=new_name,
+        referent_table=referent_table,
+        local_cols=local_cols,
+        remote_cols=remote_cols,
+        ondelete=ondelete,
+    )
+
+
+def upgrade():
+    """Rename dag_schedule_dataset_alias_reference constraint."""
+    with op.batch_alter_table("dag_schedule_dataset_alias_reference", schema=None) as batch_op:
+        _rename_fk_constraint(
+            batch_op=batch_op,
+            original_name="dsdar_dataset_fkey",
+            new_name="dsdar_dataset_alias_fkey",
+            referent_table="dataset_alias",
+            local_cols=["alias_id"],
+            remote_cols=["id"],
+            ondelete="CASCADE",
+        )
```

Comment:
Unfortunately, we will have to do a little more here @Lee-W because we already have the case where the constraint is named differently for 2 different sets of users:

It is named
1. `dsdar_dataset_fkey `: for 2.8.0 -> 2.10.2: because they are created from migration
2. `dsdar_dataset_alias_fkey`: for new install 2.10.2: because they are created from model

So you will have to do similar if/else condition as you did on L85. Otherwise this might result in constraint not found error for folks with Case (2) above

Reply:
Thanks! Just updated it